### DISCUSSION TITLE
Could not load file or assembly 'System.Threading.AccessControl' after update to 5.0.0

### DISCUSSION THREAD

**Discussion by silentmike-m** [JasperFx/marten#2188] [General]
After update to Marten 5.0.0 I had to manually install System.Threading.AccessControl nuget package.
Otherwise I would get an exception:

```csharp
Generated code for DocumentStorage.CoreEntityProvider1049694721
Could not make an assembly reference to mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089

System.IO.FileNotFoundException: Could not load file or assembly 'System.Threading.AccessControl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Nie można odnaleźć określonego pliku.
File name: 'System.Threading.AccessControl, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at System.Reflection.RuntimeAssembly.InternalLoad(ObjectHandleOnStack assemblyName, ObjectHandleOnStack requestingAssembly, StackCrawlMarkHandle stackMark, Boolean throwOnFileNotFound, ObjectHandleOnStack assemblyLoadContext, ObjectHandleOnStack retAssembly)
   at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, RuntimeAssembly requestingAssembly, StackCrawlMark& stackMark, Boolean throwOnFileNotFound, AssemblyLoadContext assemblyLoadContext)
   at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
   at LamarCompiler.AssemblyGenerator.ReferenceAssembly(Assembly assembly)
```

Despite this error the data was read.

**Comment by jeremydmiller** [comment]
Okay, but what are you expecting to happen here? I don't think this is an issue in Marten or event LamarCompiler. Something in your codebase says that it depends on that System.Threading.AccessControl assembly, but Nuget didn't put it there for whatever reason. Think you might have had a conflict on that assembly version maybe.

**Comment by jannikbuschke** [comment]
Seing the same, adding an explicit reference also solved it.

**Reply by cocowalla** [reply]
I had the same issue a while back, but it only happened when using Marten's auto type generation. I wonder if it's somehow linked to that?

**Reply by jeremydmiller** [reply]
It is related to the codegen, but I'll just say the same thing yet again. Something in your application declares a transitive dependency on that missing assembly, but yet it's not getting pulled in via Nuget for who knows why. Adding the explicit reference should solve it.
